gatsby-node.js

```diff
@@ -1,168 +1,2 @@
-        );
-    });
-};
-
-exports.onCreateNode = async ({ node }) => {
-    if (node.internal.type === 'Asset') {
-        assets.push(node.id);
-    }
-};
-
-const filteredPageGroups = allSeries => {
-    // featured articles are in pageGroups but not series, so we remove them
-    homeFeaturedArticles = allSeries.home;
-    learnFeaturedArticles = allSeries.learn;
-    // also remove a group of excluded articles
-    excludedLearnPageArticles = allSeries.learnPageExclude;
-    delete allSeries.home;
-    delete allSeries.learn;
-    delete allSeries.learnPageExclude;
-    return allSeries;
-};
-
-exports.createPages = async ({ actions, graphql }) => {
-    const { createPage } = actions;
-    const [, metadata, result] = await Promise.all([
-        saveAssetFiles(assets, stitchClient),
-        stitchClient.callFunction('fetchDocument', [
-            DB,
-            METADATA_COLLECTION,
-            constructDbFilter(PAGE_ID_PREFIX),
-        ]),
-        graphql(articles),
-    ]);
-
-    if (result.error) {
-        throw new Error(`Page build error: ${result.error}`);
-    }
-
-    const allSeries = filteredPageGroups(metadata.pageGroups);
-
-    result.data.allArticle.nodes.forEach(article => {
-        createArticlePage(
-            article.slug,
-            slugContentMapping,
-            allSeries,
-            metadata,
-            createPage
-        );
-    });
-
-    const tagTypes = ['author', 'languages', 'products', 'tags', 'type'];
-    const tagPages = tagTypes.map(type =>
-        createTagPageType(
-            type,
-            createPage,
-            metadata,
-            slugContentMapping,
-            stitchClient
-        )
-    );
-    await Promise.all(tagPages);
-};
-
-// Prevent errors when running gatsby build caused by browser packages run in a node environment.
-exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
-    if (stage === 'build-html') {
-        actions.setWebpackConfig({
-            module: {
-                rules: [
-                    {
-                        test: /mongodb-stitch-browser-sdk/,
-                        use: loaders.null(),
-                    },
-                ],
-            },
-        });
-    }
-    actions.setWebpackConfig({
-        resolve: {
-            alias: {
-                // Use noop file to prevent any preview-setup errors
-                previewSetup: path.resolve(__dirname, 'preview/noop.js'),
-                useSiteMetadata: path.resolve(
-                    __dirname,
-                    'src/hooks/use-site-metadata.js'
-                ),
-            },
-        },
-    });
-};
-
-exports.onCreatePage = async ({ page, actions }) =>
-    onCreatePage(
-        page,
-        actions,
-        stitchClient,
-        homeFeaturedArticles,
-        learnFeaturedArticles,
-        excludedLearnPageArticles
-    );
+require = require('esm')(module);
+module.exports = require('./gatsby-node.esm.js');
```

Comment:
Here is the `esm` magic, suggested in [this Gatsby issue thread](https://github.com/gatsbyjs/gatsby/issues/7810#issuecomment-449741977)